rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -407,7 +421,20 @@ private void initStatus(WorkflowRun run) {
         } else if (execution.getCauseOfFailure() != null) {
             setStatus(StatusExt.valueOf(execution.getCauseOfFailure()));
         } else if (execution.isComplete()) {
-            setStatus(StatusExt.SUCCESS);
+            Result r = run.getResult();
+            if (r == Result.NOT_BUILT) {
```

Comment:
surely this should be using `Result.isWorseThan()` - at least for a certain class of Results?